exp/addons/internal/controllers/clusterresourceset_controller.go

```diff
@@ -312,15 +319,14 @@ func (r *ClusterResourceSetReconciler) ApplyClusterResourceSet(ctx context.Conte
 			errList = append(errList, err)
 		}
 
-		// Since maps are not ordered, we need to order them to get the same hash at each reconcile.
 		data, ok := unstructuredObj.UnstructuredContent()["data"]
 		if !ok {
 			errList = append(errList, errors.New("failed to get data field from the resource"))
 			continue
 		}
 
 		// now calculate the new hash to compare with the old
-		dataList, newHash := getDataListAndHash(unstructuredObj.GetKind(), data.(map[string]interface{}), errList)
+		dataList, newHash := getDataListAndHash(unstructuredObj.GetKind(), data.(map[string]interface{}), &errList)
```

Comment:
Another option could be to return errors and append after, tbh not sure whats more conventional.
```suggestion
		dataList, newHash, errDataListAndHash := getDataListAndHash(unstructuredObj.GetKind(), data.(map[string]interface{}))
		errList = append(errList, errDataListAndHash)
```

Reply:
I used your idea. Thanks!